src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,40 @@
 
 class Animal {
   // write your code here
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
   // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+
+  bite(herbivor) {
+    if (herbivor instanceof Herbivore && herbivor.hidden === false) {
+      herbivor.health -= 50;
+    }
+
+    if (herbivor.health === 0) {
```

Comment:
Ofc in this case when we have only damage 50 there's no chance that the health is going to be less than zero, but if we decide to change the damage of the `bite()` method to let's say 27 - the animal health will be less than zero and the animal won't be removed in this case